software-engineering-policies/SourceControl/BranchProtectionPolicy.md

```diff
@@ -0,0 +1,46 @@
+# Branch Protection
+
+Branch protection on active repositories is important to ensure that we are doing the best we can to be clean and secure. The policies set out below are required for your main branch but options for other branches.
+
+The following options should be selected. An explanation is provided where necessary as to why this should be enabled.
+
+## Require a pull request before merging
+
+We should never be making any code changes to the main branch without making a pull request. This can lead to a messy history, conflicts and broken builds that can cause outages. By performing a pull request, your code can be put through quality checks, code review and even be run in an environment before being merged.
+
+### Require approvals
+
+The team size will dictate the number of approvals required to move a pull request through the process but the ideal here should be 2. This allows for multiple views of the code from different angles to make sure what you are trying to merge fits what is required. However it is understood sometimes team sizes are not large enough for this so an absolute minimum would be 1 reviewer.
+
+### Dismiss stale pull request approvals when new commits are pushed
+
+If a code change is made, no matter how tiny, it is best that your reviewers take another look just to be sure that everything is ok. It may seem overkill for something like a line deletion or missing semicolon however it stops code creep and ensures you are proofing your code prior to reviews in the future.
+
+### Require reviews from Code Owners
+
+The code owners should be the gate keepers of the repository and no code should pass this point without their knowledge or approval. It is the code owners who have the knowledge of the code and its purpose that can help guide a PR so having this is extra important.
+
+
+## Require status checks to pass before merging
+
+Status checks are the automated checks that run on the code to support in quality and security. Requiring these is a must but it is not enough to just turn this on, you need to be explicit in which checks you want requiring.
+
+### Require branches to be up to date before merging
+
+It is pointless running checks and doing code reviews on code that is not inline with your main branch. It could be that your code is fine however there has been a change made that conflicts with yours that would not be found until merging. This check allows us to prevent this from happening.
+
+### Snyk SAST status check to be required
+
+Snyk has been setup against all Repos so when you type `snyk` into the text box it should hopefully show up and you can select it. It is important that we have this as a required check as we do not want to push insecure code into main. 
```

Comment:
"you type `snyk` into the text box" - what does this mean?

Reply:
Hopefully this is clearer now, if not let me know and I might provide supporting screenshots